### ISSUE TITLE
Broken Add New Entry via DOI search

### ISSUE THREAD

**Issue by DOFfactory** [JabRef/jabref#4685]
JabRef 5.0-dev--snapshot--2019-02-20--master--3589dc3cb
Windows 10 10.0 amd64 
Java 1.8.0_172

1. Library -> New Entry
2. Select DOI -> insert 10.2139/ssrn.567842
3. hit enter ->   Error: Uncaught exception occurred in Thread[JavaFX Application Thread,5,main]

<details>
  <summary>Log File</summary>

  ```
java.lang.IndexOutOfBoundsException: Index: 3, Size: 3
	at java.util.ArrayList.rangeCheck(Unknown Source)
	at java.util.ArrayList.get(Unknown Source)
	at java.awt.Container.createHierarchyEvents(Unknown Source)
	at java.awt.Container.createHierarchyEvents(Unknown Source)
	at java.awt.Container.createHierarchyEvents(Unknown Source)
	at java.awt.Container.createHierarchyEvents(Unknown Source)
	at java.awt.Container.createHierarchyEvents(Unknown Source)
	at java.awt.Dialog.conditionalShow(Unknown Source)
	at java.awt.Dialog.show(Unknown Source)
	at java.awt.Component.show(Unknown Source)
	at java.awt.Component.setVisible(Unknown Source)
	at java.awt.Window.setVisible(Unknown Source)
	at java.awt.Dialog.setVisible(Unknown Source)
	at org.jabref.gui.JabRefDialog.setVisible(JabRefDialog.java:45)
	at org.jabref.gui.EntryTypeViewModel.lambda$runFetcherWorker$300(EntryTypeViewModel.java:147)
	at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:49)
	at javafx.event.Event.fireEvent(Event.java:198)
	at javafx.concurrent.EventHelper.fireEvent(EventHelper.java:219)
	at javafx.concurrent.Task.fireEvent(Task.java:1356)
	at javafx.concurrent.Task.setState(Task.java:723)
	at javafx.concurrent.Task$TaskCallable.lambda$call$501(Task.java:1434)
	at com.sun.javafx.application.PlatformImpl.lambda$null$172(PlatformImpl.java:295)
	at java.security.AccessController.doPrivileged(Native Method)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$173(PlatformImpl.java:294)
	at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
	at com.sun.glass.ui.win.WinApplication._enterNestedEventLoopImpl(Native Method)
	at com.sun.glass.ui.win.WinApplication._enterNestedEventLoop(WinApplication.java:204)
	at com.sun.glass.ui.Application.enterNestedEventLoop(Application.java:511)
	at com.sun.glass.ui.EventLoop.enter(EventLoop.java:107)
	at com.sun.javafx.tk.quantum.QuantumToolkit.enterNestedEventLoop(QuantumToolkit.java:590)
	at javafx.stage.Stage.showAndWait(Stage.java:474)
	at javafx.scene.control.HeavyweightDialog.showAndWait(HeavyweightDialog.java:162)
	at javafx.scene.control.Dialog.showAndWait(Dialog.java:341)
	at org.jabref.gui.actions.NewEntryAction.execute(NewEntryAction.java:55)
	at org.jabref.gui.actions.JabRefAction.lambda$new$816(JabRefAction.java:29)
	at org.controlsfx.control.action.Action.handle(Action.java:419)
	at org.controlsfx.control.action.Action.handle(Action.java:64)
	at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:49)
	at javafx.event.Event.fireEvent(Event.java:198)
	at javafx.scene.control.MenuItem.fire(MenuItem.java:462)
	at com.sun.javafx.scene.control.skin.ContextMenuContent$MenuItemContainer.doSelect(ContextMenuContent.java:1405)
	at com.sun.javafx.scene.control.skin.ContextMenuContent$MenuItemContainer.lambda$createChildren$343(ContextMenuContent.java:1358)
	at com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:218)
	at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.event.Event.fireEvent(Event.java:198)
	at javafx.scene.Scene$MouseHandler.process(Scene.java:3757)
	at javafx.scene.Scene$MouseHandler.access$1500(Scene.java:3485)
	at javafx.scene.Scene.impl_processMouseEvent(Scene.java:1762)
	at javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2494)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:394)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:295)
	at java.security.AccessController.doPrivileged(Native Method)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$353(GlassViewEventHandler.java:432)
	at com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:389)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:431)
	at com.sun.glass.ui.View.handleMouseEvent(View.java:555)
	at com.sun.glass.ui.View.notifyMouse(View.java:937)
	at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at com.sun.glass.ui.win.WinApplication.lambda$null$147(WinApplication.java:177)
	at java.lang.Thread.run(Unknown Source)
  ```
</details>

**Comment by AEgit** [comment]
Cannot confirm for:

JabRef 5.0-dev--snapshot--2019-02-20--master--3589dc3cb
Windows 10 10.0 amd64 
Java 1.8.0_201

Using the DOI you provided I am able to generate a new entry following your steps.

**Comment by DOFfactory** [comment]
Then we have a problem, as I do get an error (see screenshot attached). I added a logfile to my previous post for reference
![error](https://user-images.githubusercontent.com/10846169/53171649-aa265d00-35e3-11e9-8bf8-96bd6cd12c3d.png)

**Comment by DOFfactory** [comment]
I can see the only difference between our apps is the Java version. I use the Java version included in the installer, is your Java a separate install? That might be the issue possibly?

**Comment by AEgit** [comment]
My Java is, indeed, a separate install.

**Comment by Siedlerchr** [comment]
Try the latest jre/jdk 202:
Works fine for me.
https://www.oracle.com/technetwork/java/javase/downloads/jre8-downloads-2133155.html

Might also be a duplicate of #4659

**Comment by Siedlerchr** [comment]
Btw, you can directly paste a doi from your clipbaord on your maintable by pressing <kbd>ctrl</kbd>+<kbd>v</kbd>

**Comment by Siedlerchr** [comment]
The import dialog has been reworked and converted to javafx as well. Could you please test again with the latest master version?

**Comment by DOFfactory** [comment]
thanks @Siedlerchr, looks very nice. I really like all the other UI changes. I tested my search against Google Scholar, and it ends in a never-ending "Processing ...". ACM worked, I haven't tested others. I kinda liked the compact view we had previously, now there is too much scrolling needed to find this or that. IMHO a search within the result window would simply things in this regard. What do you think?

**Comment by Siedlerchr** [comment]
@DOFfactory  Google Scholar is difficult and often blocked by Google, there is not much we can do about.
A search sounds like a good idea. Just open a new issue/enhancement e.g. Search in import (inspection) window
